pkg/sql/schemachanger/scexec/scmutationexec/eventlog.go

```diff
@@ -110,8 +110,9 @@ func asEventPayload(
 		}
 		switch targetStatus {
 		case scpb.Status_PUBLIC:
-			return &eventpb.AlterTable{
+			return &eventpb.CreateIndex{
```

Comment:
This makes sense in and of itself (and I prefer it) but have you checked what the legacy schema changer does? Making this an eventpb.AlterTable may have been deliberate.